clients/src/main/java/org/apache/kafka/common/record/FileLogInputStream.java

```diff
@@ -56,9 +57,7 @@ public FileChannelLogEntry nextEntry() throws IOException {
             return null;
 
         logHeaderBuffer.rewind();
-        channel.read(logHeaderBuffer, position);
-        if (logHeaderBuffer.hasRemaining())
-            return null;
+        Utils.readFullyOrFail(channel, logHeaderBuffer, position);
```

Comment:
@junrao, is it OK to throw an exception instead of returning `null` if we reach EOF before filling the buffer here?